mitiq/pec/representations/tests/test_learning.py

```diff
@@ -205,48 +239,121 @@ def noisy_execute_qiskit(circ: qiskit.QuantumCircuit) -> float:
 
 
 @pytest.mark.parametrize("epsilon", [0.05, 0.1])
-@pytest.mark.parametrize("operations", [CNOT_ops[0]])
-# We assume the operation "op" appears just once in the circuit such
-# that it's enough to add a single noise channel after that operation.
-def test_learn_depolarizing_noise_parameter(epsilon, operations):
+def test_learn_depolarizing_noise_parameter(epsilon):
     """Test the learning function with initial noise strength with a small
     offset from the simulated noise model values"""
 
-    index = operations[0]
-    op = operations[1]
-    offset = 0.1
-    eta = 0
+    operations_to_learn = [Circuit(op[1]) for op in CNOT_ops]
 
-    pec_kwargs_learning = {"num_samples": 300, "random_state": 1}
+    offset = 0.1
 
     def noisy_execute(circ: Circuit) -> np.ndarray:
         noisy_circ = circ.copy()
-        qubits = op.qubits
-        for q in qubits:
-            noisy_circ.insert(
-                index + 1,
-                biased_noise_channel(epsilon, eta)(q),
-                strategy=InsertStrategy.EARLIEST,
-            )
+        insertions = []
+        for op in CNOT_ops:
+            index = op[0] + 1
+            qubits = op[1].qubits
+            for q in qubits:
+                insertions.append((index, DepolarizingChannel(epsilon)(q)))
+        noisy_circ.batch_insert(insertions)
         return ideal_execute(noisy_circ)
 
     noisy_executor = Executor(noisy_execute)
 
     epsilon0 = (1 - offset) * epsilon
 
-    operations_to_learn = [Circuit(operations[1])]
+    pec_data = np.loadtxt(
+        "./mitiq/pec/representations/tests/learning_pec_data/"
+        + "learning_pec_data_eps_"
+        + str(epsilon).replace(".", "_")
+        + ".txt"
+    )
```

Comment:
We really shouldn't join strings by hand to make filepaths. Check out the [`os.path.join`](https://docs.python.org/3/library/os.path.html#os.path.join) function in the standard lib.

Also I think string concatenation with `+` is very slow, and hence it is generally recommended to use f-Strings instead.

Reply:
Thanks for linking to `os.path.join` I wasn't aware of it. Not sure if I fully captured the intended change, but I think it's better now.